onnx/defs/tensor/old.cc

```diff
@@ -396,6 +396,65 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
+static const char* Upsample_ver9_doc = R"DOC(
+Upsample the input tensor.
+Each dimension value of the output tensor is:
+  output_dimension = floor(input_dimension * scale).
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    Upsample,
+    9,
+    OpSchema()
+        .Attr(
+            "mode",
+            "Two interpolation modes: nearest (default), and linear (including bilinear, trilinear, etc)",
+            AttributeProto::STRING,
+            std::string("nearest"))
+        .Input(0, "X", "N-D tensor", "T")
+        .Input(
+            1,
+            "scales",
+            "The scale array along each dimension. It takes value greater than or equal to 1."
+            " The number of elements of 'scales' should be the same as the rank of input 'X'.",
+            "tensor(float)")
+        .Output(0, "Y", "N-D tensor after resizing", "T")
+        .TypeConstraint(
+            "T",
+            OpSchema::all_tensor_types(),
+            "Constrain input 'X' and output 'Y' to all tensor types.")
+        .SetDoc(Upsample_ver9_doc)
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          if (!hasNInputShapes(ctx, 1)) {
+            return;
+          }
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
+          auto& input_shape = getInputShape(ctx, 0);
+          auto* output_shape = getOutputShape(ctx, 0);
+          output_shape->clear_dim();
```

Comment:
Can we check whether output_shape already has enough information? 

If so, we don't need to erase the data. Instead, we just finish the shape inference here.

Reply:
@houseroad  I'd suggest the check happens outside when running shape inference, otherwise, this kind of check needs happening for every shape inference function implementation. Make sense?